### PR TITLE
Add support for defining labels in the component-descriptor trait

### PR DESCRIPTION
**Pull request by AndreasBurger**

### CODE REVIEW

File: concourse/model/traits/component_descriptor.py [hunk 1]
```diff
@@ -83,6 +85,12 @@
             If not configured, the CICD-landscape's default ctx will be used.
         '''
     ),
+    AttributeSpec.optional(
+        name='component_descriptor_labels',
```
Comment (ccwienk) [inline concourse/model/traits/component_descriptor.py:89]: I would prefer to call it `component_labels` - it is already below the `component_descriptor_trait` (DRY)

File: concourse/model/traits/component_descriptor.py [hunk 2]
```diff
@@ -83,6 +85,12 @@
...
+        default=(),
+        type=list,
```
Comment (ccwienk) [inline concourse/model/traits/component_descriptor.py:91]: `typing.List[cm.Label]`?

File: concourse/steps/component_descriptor.mako [hunk 3]
```diff
@@ -45,6 +45,7 @@ with open(version_file_path) as f:
   effective_version = f.read().strip()
 
 component_name = '${descriptor_trait.component_name()}'
+component_labels = ${descriptor_trait.component_descriptor_labels()}
```
Comment (ccwienk) [inline concourse/steps/component_descriptor.mako:48]: I think I already asked that: will this work if the attr is not specified (as in: will the result be valid yaml)? caveat (sorry for "tricking" you into this: I think, the tuple literal is not compatible w/ yaml, is it?

**Review by ccwienk — CHANGES_REQUESTED** [review]

**Review by jkrayl — APPROVED** [review]

File: concourse/steps/component_descriptor.mako (hunk 3 above)
Comment (AndreasBurger) [reply concourse/steps/component_descriptor.mako:48]: Yes, pyyaml will dump tuples as lists, but when using `safe_dump` the type-information is lost. On deserialisation the labels will be in a list.

Comment (ccwienk) [reply concourse/steps/component_descriptor.mako:48]: I just tried:
```>>> yaml.dump({'a': ()})
'a: !!python/tuple []\n'
```

so I figure pyyaml will retain the tuple-metadata. As the component-descriptor-spec requires a _list_, I think we should also actually dump an (empty) list

File: concourse/steps/component_descriptor.py [hunk 4]
```diff
@@ -81,7 +82,7 @@ def base_component_descriptor_v2(
         ],
         componentReferences=[], # added later
         resources=[], # added later
-        labels=[], # added later
+        labels=component_labels,
```
Comment (ccwienk) [inline concourse/steps/component_descriptor.py]: if this attr is to be modified later, defaulting it to a tuple is rather unhandy (I already changed this to a list, though)..

**Review by ccwienk — APPROVED** [review]

**Pull request merged**